test/error_tpm2-tss-engine-common.c

```diff
@@ -21,7 +21,83 @@ __wrap_Esys_Initialize()
     backtrace_symbols_fd(b, backtrace(b, sizeof(b)/sizeof(b[0])), STDOUT_FILENO);
     return -1;
 }
-    
+
+void
+check_init_auth(void **state)
+{
+    (void)(state);
+    int i;
+    TPM2B_AUTH auth = { .size = -1 };
+
+    i = init_auth(&auth, NULL, 0);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 0);
+
+
+    i = init_auth(&auth, "abc", 0);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 3);
+    assert_memory_equal(auth.buffer, "abc", auth.size);
+
+    i = init_auth(&auth, "defghi", 3);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 3);
+    assert_memory_equal(auth.buffer, "def", auth.size);
+
+    i = init_auth(&auth, "\xDE\xAD\xBE\xEF", 3);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 3);
+    assert_memory_equal(auth.buffer, "\xDE\xAD\xBE", auth.size);
+
+
+    i = init_auth(&auth, "str:abc", 0);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 3);
+    assert_memory_equal(auth.buffer, "abc", auth.size);
+
+    i = init_auth(&auth, "str:defghi", 7);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 7);
+    assert_memory_equal(auth.buffer, "str:def", auth.size);
+
+    i = init_auth(&auth, "str:str:abc", 0);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 7);
+    assert_memory_equal(auth.buffer, "str:abc", auth.size);
+
+    i = init_auth(&auth, "str:str:defghi", 7);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 7);
+    assert_memory_equal(auth.buffer, "str:str", auth.size);
+
+    i = init_auth(&auth, "str:hex:DEADBEEF", 0);
+    assert_int_equal(i, 1);
+    assert_true(auth.size == 12);
+    assert_memory_equal(auth.buffer, "hex:DEADBEEF", auth.size);
+
+    i = init_auth(&auth, "str:hex:DEADBEEF", 10);
```

Comment:
mix up the strings a bit, you only try deadbeef here and throughout.

Also, what about over-length passwords?